### PR TITLE
What, behind the rabbit? It is the rabbit. (Added Killer Bunnies)

### LINKED ISSUE #265: Add Killer Bunnies
Add killer bunnies to the game.

### LINKED ISSUE #266: Carrots
Add carrots to the game

### LINKED ISSUE #267: Bait
Implemented bait (carrots) and implement that enemies are distracted temporarily by bait (carrots).

### PR DESCRIPTION
**Pull request by ChielBruin**
Added killer bunnies to the game.
The bunnies will hop around and walk towards the closest player or carrot.

Closes #265, closes #266 and closes #267.
Note: they do not yet kill the player as health is not implemented, but the code is there that should work when it is implemented.

### CODE REVIEW

File: src/main/java/nl/tudelft/contextproject/controller/GameController.java [hunk 1]
```diff
@@ -130,6 +130,13 @@ protected void attachLevel() {
 		AmbientLight al = new AmbientLight();
 		al.setColor(ColorRGBA.White.mult(.5f));
 		addLight(al);
+		
+//		Entity e = new KillerBunny(new Vector3f(4, 2, 4));
+//		game.getEntities().add(e);
+//		
+//		Entity c = new Carrot();
+//		c.move(4, 1, 4);
+//		game.getEntities().add(c);
```
Comment (Geweldig) [inline src/main/java/nl/tudelft/contextproject/controller/GameController.java]: Is this needed?

Comment (ChielBruin) [inline src/main/java/nl/tudelft/contextproject/controller/GameController.java]: no it is not

File: src/test/java/nl/tudelft/contextproject/model/entities/EnemyTest.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	@Override
+	public Entity getEntity() {
+		return getEnemy();
+	}
+
+	/**
+	 * Getter for an {@link Enemy}.
+	 * 
+	 * @return
+	 * 		a fresh instance of {@link Enemy}.
```
Comment (Geweldig) [inline src/test/java/nl/tudelft/contextproject/model/entities/EnemyTest.java]: . should be removed

File: src/main/java/nl/tudelft/contextproject/model/entities/Enemy.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ */
+public abstract class Enemy extends Entity {
+
+	private EntityAI ai;
+
+	/**
+	 * Constructor that attaches the {@link EntityAI} to the entity.
+	 * 
+	 * @param ai
+	 * 		the {@link EntityAI} for this entity.
```
Comment (Geweldig) [inline src/main/java/nl/tudelft/contextproject/model/entities/Enemy.java]: . should be removed

File: src/main/java/nl/tudelft/contextproject/model/entities/Enemy.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+	public Enemy(EntityAI ai) {
+		this.ai = ai;
+		this.ai.setOwner(this);
+	}
+	
+	/**
+	 * Set the ai of this {@link Enemy}.
+	 * 
+	 * @param ai
+	 * 		the new ai of this instance.
```
Comment (Geweldig) [inline src/main/java/nl/tudelft/contextproject/model/entities/Enemy.java]: . should be removed

File: src/main/java/nl/tudelft/contextproject/model/entities/Enemy.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+	 * @return
+	 * 		the old ai of this instance.
```
Comment (Geweldig) [inline src/main/java/nl/tudelft/contextproject/model/entities/Enemy.java]: . should be removed

File: src/main/java/nl/tudelft/contextproject/model/entities/FoodItem.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+	 * @param amount
+	 * 		the amount that was eaten
+	 */
+	public void eat(Entity eatenBy, float amount);
+
+	/**
+	 * Get the remaining amount.
+	 * 
+	 * @return
+	 * 		the remaining amount.
```
Comment (Geweldig) [inline src/main/java/nl/tudelft/contextproject/model/entities/FoodItem.java]: . should be removed

File: src/main/java/nl/tudelft/contextproject/model/entities/ai/EntityAI.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+	 * @param tpf
+	 * 		the time per frame
+	 */
+	void move(float tpf);
+
+	/**
+	 * Set the owner of this ai.
+	 * 
+	 * @param owner
+	 * 		the owner of this ai.
```
Comment (Geweldig) [inline src/main/java/nl/tudelft/contextproject/model/entities/ai/EntityAI.java]: . should be removed

**Comment by Geweldig** [comment]
Other than my complaints I think this is good to merge!

**Comment by Diocruel** [comment]
Good to be merged!

**Pull request merged**